### ISSUE TITLE
Accessing req.routeConfig in frameworkErrors handler throws TypeError

### ISSUE THREAD

**Issue by sergburn** [fastify/fastify#4807]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

4.17.0

### Node.js version

16.x

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

11.7

### Description

An attempt to get a value of `request.routeConfig` in `frameworkErrors` handler throws an exception:
```
test/router-options.test.js 2>       return this[kRouteContext][kPublicRouteContext].config
test/router-options.test.js 2>                                                       ^
test/router-options.test.js 2> TypeError: Cannot read properties of undefined (reading 'config')
```

### Steps to Reproduce

```js
test('Should not throw on access to routeConfig frameworkErrors handler - FST_ERR_BAD_URL', t => {
  t.plan(6)

  const fastify = Fastify({
    frameworkErrors: function (err, req, res) {
      t.ok(typeof req.id === 'string')
      t.ok(req.raw instanceof Readable)
      t.same(req.routerPath, undefined)
      t.same(req.routeConfig, undefined)
      //         ^^^
      // test/router-options.test.js 2>       return this[kRouteContext][kPublicRouteContext].config
      // test/router-options.test.js 2>                                                       ^
      // test/router-options.test.js 2> TypeError: Cannot read properties of undefined (reading 'config')
      res.send(`${err.message} - ${err.code}`)
    }
  })

  fastify.get('/test/:id', (req, res) => {
    res.send('{ hello: \'world\' }')
  })

  fastify.inject(
    {
      method: 'GET',
      url: '/test/%world'
    },
    (err, res) => {
      t.error(err)
      t.equal(res.body, '\'/test/%world\' is not a valid url component - FST_ERR_BAD_URL')
    }
  )
})
```

### Expected Behavior

Should return `undefined` as, for example, `req.routerPath` does

**Comment by sergburn** [comment]
If this is accepted as a valid issue, which requires a fix, I can work on a pull request.

**Comment by gurgunday** [comment]
Yeah I think the getter should return undefined instead of throwing

**Comment by ShivangRawat30** [comment]
can I work on this issue ??

**Comment by metcoder95** [comment]
Go for it 👍

**Comment by ShivangRawat30** [comment]
I looked at the code and figured that if either kRouteContext or kPublicRouteContext is undefined this will result in a TypeError at runtime, the req.routerPath is returning undefined that means kRouteContext is not undefined as 
routerPath: {
    get () {
      return this[kRouteContext].config.url
    }
  },
so I think the problem lies in kPublicRouteContext.
I am new to contributing to open source, can you help me with this ?

**Comment by metcoder95** [comment]
>If for routerPath it is returning undefined, this[kRouteContext] might be undefined as well, and that's why this[kRouteContext][kPublicRouteContext].config is throwing as TypeError,
or the problem might be in kPublicRouteContext.

It should be a matter of treating with care the returning statement.
e.g. you can use the optional chaining to return `undefined` if a given value is not present
```js
console.log(this[kRouteContext]?[kPublicRouteContext]); // Will return undefined if `kRouteContext` is undefined
```

**Comment by ShivangRawat30** [comment]
@sergburn I am having hard time Reproducing the Error can you please tell me where you wrote the above code.

**Comment by sergburn** [comment]
@ShivangRawat30 put it to `tests/router_options.test.js` and add `const { Readable } = require('stream')` at the top

**Comment by giuliowaitforitdavide** [comment]
I provided a fix for this bug in #4826 
Sorry @ShivangRawat30 , I didn't read you were already working on that

**Comment by metcoder95** [comment]
> @metcoder95
> 
> ```js
> console.log(this[kRouteContext][kPublicRouteContext])  //is undefined but when i logged 
> console.log(this[kRouteContext]) //it is logging [object, Object]
> ```
> 
> which means an object is present in `this[kRouteContext]` but is unable to get the value from `[kPublicRouteContext]`

That means that `kRouteContext` does not contains any value, for instance when trying to access `config` prop it will throw as we are trying to access a nonexistent value
